src/azul/chalice.py

```diff
@@ -504,34 +498,104 @@ def swagger_resource(self, file) -> Response:
                                 headers={'Content-Type': content_type},
                                 body=body)
 
-    def threshold(self, *, errors: int, throttles: int):
-        def wrapper(f):
+    @attr.define(frozen=True, kw_only=True)
+    class lambda_decorator:
```

Comment:
```suggestion
    class LambdaDecorator:
```

and should be abstract because it is not usable as a decorator due the lack of the `__call__` method.